plugins/outputs/arc/arc_test.go

```diff
@@ -0,0 +1,467 @@
+	}
+
+	err := a.Init()
+	require.NoError(t, err)
+
+	err = a.Connect()
+	require.NoError(t, err)
+
+	metrics := []telegraf.Metric{
+		testutil.MustMetric(
+			"cpu",
+			map[string]string{"host": "server01"},
+			map[string]interface{}{"usage_idle": float64(95.5)},
+			time.Now(),
+		),
+	}
+
+	err = a.Write(metrics)
+	require.Error(t, err)
+	require.Contains(t, err.Error(), "returned status 500")
+}
+
+func TestMessagePackEncoding(t *testing.T) {
+	metrics := []telegraf.Metric{
+		testutil.MustMetric(
+			"cpu",
+			map[string]string{
+				"host":        "server01",
+				"cpu":         "cpu0",
+				"environment": "production",
+			},
+			map[string]interface{}{
+				"usage_idle":   float64(95.5),
+				"usage_system": float64(2.5),
+				"usage_user":   float64(2.0),
+			},
+			time.Unix(1633024800, 0),
+		),
+		testutil.MustMetric(
+			"cpu",
+			map[string]string{
+				"host":        "server02",
+				"cpu":         "cpu1",
+				"environment": "production",
+			},
+			map[string]interface{}{
+				"usage_idle":   float64(85.0),
+				"usage_system": float64(4.5),
+				"usage_user":   float64(10.5),
+			},
+			time.Unix(1633024801, 0),
+		),
+	}
+
+	// Build columnar data
+	columns := make(map[string]interface{})
+
+	// Time column
+	columns["time"] = []int64{
+		metrics[0].Time().UnixMilli(),
+		metrics[1].Time().UnixMilli(),
+	}
+
+	// Tag columns
+	columns["host"] = []string{"server01", "server02"}
+	columns["cpu"] = []string{"cpu0", "cpu1"}
+	columns["environment"] = []string{"production", "production"}
+
+	// Field columns
+	columns["usage_idle"] = []interface{}{float64(95.5), float64(85.0)}
+	columns["usage_system"] = []interface{}{float64(2.5), float64(4.5)}
+	columns["usage_user"] = []interface{}{float64(2.0), float64(10.5)}
+
+	// Create data as map[string]interface{} for msgp
+	dataMap := map[string]interface{}{
+		"m":       "cpu",
+		"columns": columns,
+	}
+
+	// Marshal to MessagePack using tinylib/msgp
+	var buf bytes.Buffer
+	writer := msgp.NewWriter(&buf)
+	err := writer.WriteIntf(dataMap)
+	require.NoError(t, err)
+	err = writer.Flush()
+	require.NoError(t, err)
+
+	data := buf.Bytes()
+	require.NotEmpty(t, data)
+
+	// Verify it can be unmarshaled
+	reader := msgp.NewReader(bytes.NewReader(data))
+	decodedIntf, err := reader.ReadIntf()
+	require.NoError(t, err)
+
+	decoded := decodedIntf.(map[string]interface{})
+	require.Equal(t, "cpu", decoded["m"])
+	require.NotEmpty(t, decoded["columns"])
+
+	// Verify time column
+	columnsMap := decoded["columns"].(map[string]interface{})
+	timeCol, ok := columnsMap["time"]
+	require.True(t, ok)
+	timeArray, ok := timeCol.([]interface{})
+	require.True(t, ok)
+	require.Len(t, timeArray, 2)
+
+	// Verify field columns
+	usageIdleCol, ok := columnsMap["usage_idle"]
+	require.True(t, ok)
+	usageIdleArray, ok := usageIdleCol.([]interface{})
+	require.True(t, ok)
+	require.InDelta(t, 95.5, usageIdleArray[0], 0.01)
+	require.InDelta(t, 85.0, usageIdleArray[1], 0.01)
+}
```

Comment:
Please don't do this. Instead write a bunch of metrics and then check the received data by unmarshalling the received data and check against an expectation!